src/main/java/ui/UserInterface.java

```diff
@@ -92,24 +106,42 @@ private void printAllBorrowedBookById () {
         List<Book> listOfBooks = libraryService.allBorrowedBookByReaderId(readerId.trim());
 
         if (listOfBooks.isEmpty()) {
-            System.out.println("This reader hasn't borrowed the book yet");
+            System.out.println("This reader hasn't borrowed the book yet.");
         } else {
             listOfBooks.forEach(System.out::println);
         }
     }
 
-    private void printCurrentReaderOfBookByReaderId () {
+    private void printCurrentReaderOfBookByBookId () {
         System.out.println("Write book id:");
         String bookId = input.nextLine();
-        Optional<Reader> currentReader = libraryService.currentReaderOfBook(bookId.trim());
 
-        if (currentReader.isEmpty()) {
-            System.out.println("This book is in the library");
+        libraryService.currentReaderOfBook(bookId.trim())
+                .ifPresentOrElse(
+                        System.out::println,
+                        () -> System.out.println("This book is in the library.")
+                );
+    }
+    private void printAllReadersWithBorrowedBooks() {
+        Map<Reader, Optional<Book>> allReadersWithBorrowedBooks = libraryService.findAllReadersWithBorrowedBooks();
+        if (allReadersWithBorrowedBooks.isEmpty()) {
+            System.out.println("There are no books borrowed.");
         } else {
-            System.out.println(currentReader.get());
+            allReadersWithBorrowedBooks.forEach((key, value) -> System.out.println(key + " : " + value.orElse(null)));
         }
     }
 
+    private void printAllBooksWithReaders() {
+        libraryService.findAllBooksWithReaders().forEach(
+                (key, value) -> {
+                    if (value.isEmpty()) {
+                        System.out.println(key + " : available");
+                    } else {
+                        System.out.println(key + " : " + value.get());
+                    }
+                });
+    }
```

Comment:
Обращай внимание на возможности `Optional` и, по возможности, используй!
На это обращают внимание, когда смотрят код потенциального джуна. Нужно чтобы ни у кого не было сомнений, что ты знаешь современный синтаксис.

```suggestion
    private void printAllBooksWithReaders() {
        libraryService.findAllBooksWithReaders().forEach(
                (key, value) ->
                    value.ifPresentOrElse(
                        reader -> System.out.println(key + " : " + reader),
                        () -> System.out.println(key + " : available")
                    )
                );
    }
```